build/do.rq

```diff
@@ -163,6 +184,131 @@ write_readme {
 	run("./regal table --write-to-readme bundle")
 }
 
+fetch_engine_caps {
+	fetch_eopa_caps
+}
+
+fetch_eopa_caps {
+
+	# git ls-remote --tags output looks like this:
+	#
+	#	...
+	#	bab5bdd4bb2de45c7d1bcf5d7b5df3c61a131f38        refs/tags/v1.0.0
+	#	bab5bdd4bb2de45c7d1bcf5d7b5df3c61a131f38        refs/tags/v1.0.1
+	#	9ace37dda9859a1afd2fcd6a87dd7f1955be4dc4        refs/tags/v1.10.1
+	#	...
+	#
+	# so .stdout will look like this:
+	#
+	#	...
+	#	[
+	#		"bab5bdd4bb2de45c7d1bcf5d7b5df3c61a131f38\trefs",
+	#		"tags",
+	#		"v1.0.0"
+	#	],
+	#	[
+	#		"bab5bdd4bb2de45c7d1bcf5d7b5df3c61a131f38\trefs",
+	#		"tags",
+	#		"v1.0.1"
+	#	],
+	#	[
+	#		"bab5bdd4bb2de45c7d1bcf5d7b5df3c61a131f38\trefs",
+	#		"tags",
+	#		"v1.1.0"
+	#	],
+	#	...
+
+	print("fetching tags for enterprise-opa repository")
+
+	eopa_tags_result := rq.run([
+			"git",
+			"ls-remote",
+			"--tags",
+			"https://github.com/styrainc/enterprise-opa"
+		], {
+			"stdout_spec": {
+				"format": "raw",
+				"options": {
+					"raw.fs": "/",
+					"raw.rs": "[\n\r]"
+				},
+			}
+		}
+	)
+
+	{ rq.error(sprintf("\nstdout: %s\nstderr: %s\n", [eopa_tags_result.stdout, eopa_tags_result.stderr])) | eopa_tags_result.exitcode != 0 }
```

Comment:
To avoid a really long line, and a fairly esoteric comprehension, I'd use a helper function like:

```rego
error_on_non_zero(result) if {
	result.exitcode != 0
	rq.error(sprintf("\nstdout: %s\nstderr: %s\n", [result.stdout, result.stderr])) 
}
```

Reply:
You also need an `else { true }`, or it will stop evaluation of the rule body it is called from.

Reply:
Aha, I thought the purpose of `rq.error` was to blow up

Reply:
It is, but the purpose of the esoteric comprehension is to make it _conditional_. With just the function as you wrote it, the happy path `exitcode == 0` case is `undefined` when you call the function directly.

Reply:
b5cb5c5